server/src/main/java/org/apache/kafka/security/authorizer/AclEntry.java

```diff
@@ -0,0 +1,209 @@
+import static org.apache.kafka.common.acl.AclOperation.WRITE;
+
+public class AclEntry extends AccessControlEntry {
+    private static final DecodeJson.DecodeInteger INT = new DecodeJson.DecodeInteger();
+    private static final DecodeJson.DecodeString STRING = new DecodeJson.DecodeString();
+
+    public static final KafkaPrincipal WILDCARD_PRINCIPAL = new KafkaPrincipal(KafkaPrincipal.USER_TYPE, "*");
+    public static final String WILDCARD_PRINCIPAL_STRING = WILDCARD_PRINCIPAL.toString();
+    public static final String WILDCARD_HOST = "*";
+    public static final String WILDCARD_RESOURCE = ResourcePattern.WILDCARD_RESOURCE;
+    public static final String RESOURCE_SEPARATOR = ":";
+    public static final Set<ResourceType> RESOURCE_TYPES = Arrays.stream(ResourceType.values())
+        .filter(t -> !(t == ResourceType.UNKNOWN || t == ResourceType.ANY))
+        .collect(Collectors.toSet());
+    public static final Set<AclOperation> ACL_OPERATIONS = Arrays.stream(AclOperation.values())
+        .filter(t -> !(t == AclOperation.UNKNOWN || t == AclOperation.ANY))
+        .collect(Collectors.toSet());
+
+    private static final String PRINCIPAL_KEY = "principal";
+    private static final String PERMISSION_TYPE_KEY = "permissionType";
+    private static final String OPERATION_KEY = "operation";
+    private static final String HOSTS_KEY = "host";
+    public static final String VERSION_KEY = "version";
+    public static final int CURRENT_VERSION = 1;
+    private static final String ACLS_KEY = "acls";
+
+    public final AccessControlEntry ace;
+    public final KafkaPrincipal kafkaPrincipal;
+
+    public AclEntry(AccessControlEntry ace) {
+        super(ace.principal(), ace.host(), ace.operation(), ace.permissionType());
+        this.ace = ace;
+
+        kafkaPrincipal = ace.principal() == null
+            ? null
+            : SecurityUtils.parseKafkaPrincipal(ace.principal());
+    }
+
+    /**
+     * Parse JSON representation of ACLs
+     * @param bytes of acls json string
+     *
+     * <p>
+        {
+            "version": 1,
+            "acls": [
+                {
+                    "host":"host1",
+                    "permissionType": "Deny",
+                    "operation": "Read",
+                    "principal": "User:alice"
+                }
+            ]
+        }
+     * </p>
+     *
+     * @return set of AclEntry objects from the JSON string
+     */
+    public static Set<AclEntry> fromBytes(byte[] bytes) throws IOException {
+        if (bytes == null || bytes.length == 0)
+            return Collections.emptySet();
+
+        Optional<JsonValue> jsonValue = Json.parseBytes(bytes);
+        if (!jsonValue.isPresent())
+            return Collections.emptySet();
+
+        JsonObject js = jsonValue.get().asJsonObject();
+
+        //the acl json version.
+        Utils.require(js.apply(VERSION_KEY).to(INT) == CURRENT_VERSION, () -> "Version must be equal to " + CURRENT_VERSION);
```

Comment:
I know this is an improvement but it is a breaking functionality too. @ijuma is this okay!

Reply:
We can do the improvement later (if need).

@nizhikov could you please revert this change?

Reply:
Done